setup.py

```diff
@@ -0,0 +1,111 @@
+import re
+import shutil
+from pathlib import Path
+
+from setuptools import find_packages, setup
+from setuptools.command.install import install
+
+
+# Remove stale QEfficient.egg-info directory to avoid issues
+stale_egg_info = Path(__file__).parent / "QEfficient.egg-info"
+if stale_egg_info.exists():
+    print(f"Removing stale {stale_egg_info}")
+    shutil.rmtree(stale_egg_info)
+
+_deps = [
+    "filelock",
+    "numpy>=1.23.0",
+    "packaging",
+    "protobuf==3.20.2",
+    "pyyaml",
+    "regex!=2019.12.17",
+    "requests",
+    "tqdm",
+    "transformers==4.44.2",
+    "torch==2.0.0+cpu",
```

Comment:
@quic-jhugo @carlstreeter-quic Please review.
To summarize the package updates:

- `transformers from v4.41.2 to v4.44.2`
- `torch==2.0.0+cpu`
- `peft==0.12.0`